### PR TITLE
Laboratory work #4, Sofya Kostina - 21FPL2

### PR DESCRIPTION
**Pull request by d07l**

### CODE REVIEW

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def fill_from_tokens(self, tokens: tuple[int, ...], window_length: int) -> None:
         for pair in extract_pairs(tokens, window_length):
             self.add_edge(*pair)
 
-    # Step 8.2
+        # Step 8.2
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:360]: revert

File: lab_3_keywords_textrank/main.py [hunk 2]
```diff
@@ -640,7 +640,6 @@ def train(self) -> None:
         vertices = self._graph.get_vertices()
         for vertex in vertices:
             self._scores[vertex] = 1.0
-
         for _ in range(0, self._max_iter):
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:643]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (d07l) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py (hunk 2 above)
Comment (d07l) [reply lab_3_keywords_textrank/main.py]: done

File: requirements.txt [hunk 3]
```diff
@@ -1 +0,0 @@
-
```
Comment (artyomtugaryov) [inline requirements.txt:1]: ? Revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 3 above)
Comment (d07l) [reply requirements.txt]: done

**Review by artyomtugaryov — APPROVED** [review]

File: lab_4_summarization_textrank/start.py [hunk 4]
```diff
@@ -27,6 +30,25 @@
...
+    matrix.fill_from_sentences(sentences)
+    summarizer = TextRankSummarizer(matrix)
+    summarizer.train()
+    RESULT = summarizer.make_summary(10)
+    #print(RESULT)
+
+    # step 11
+    buddy = Buddy(paths_to_texts, stop_words, tuple(string.punctuation), idf)
+    QUERY = "Юрий Гагарин"
+    print(buddy.reply(QUERY))
```
Comment (demid5111) [inline lab_4_summarization_textrank/start.py:51]: demonstrate and handle exceptions

**Pull request closed without merging**